pilot/pkg/workloadentry/controller.go

```diff
@@ -0,0 +1,308 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workloadentry
+
+import (
+	"context"
+	"math"
+	"strings"
+	"sync"
+	"time"
+
+	"golang.org/x/time/rate"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"istio.io/istio/pilot/pkg/features"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/workloadentry/internal/autoregistration"
+	"istio.io/istio/pilot/pkg/workloadentry/internal/health"
+	workloadentrystore "istio.io/istio/pilot/pkg/workloadentry/internal/store"
+	"istio.io/istio/pkg/config"
+	"istio.io/istio/pkg/config/schema/gvk"
+	"istio.io/istio/pkg/kube/controllers"
+	"istio.io/istio/pkg/queue"
+	istiolog "istio.io/pkg/log"
+)
+
+const (
+	// maxRetries is the number of times a service will be retried before it is dropped out of the queue.
+	// With the current rate-limiter in use (5ms*2^(maxRetries-1)) the following numbers represent the
+	// sequence of delays between successive queuings of a service.
+	//
+	// 5ms, 10ms, 20ms, 40ms, 80ms
+	maxRetries = 5
+)
+
+var log = istiolog.RegisterScope("wle", "wle controller debugging")
+
+type HealthEvent = health.HealthEvent
+
+var WorkloadControllerAnnotation = workloadentrystore.WorkloadControllerAnnotation
+
+type WorkloadEntryCleaner interface {
+	GetCleanupGracePeriod() time.Duration
+	ShouldCleanup(wle *config.Config, maxConnectionAge time.Duration) bool
+	Cleanup(wle *config.Config, periodic bool)
+}
+
+type Controller struct {
```

Comment:
nit: group vars and types together

Reply:
This PR will be replaced by https://github.com/istio/istio/pull/44834.

I've addressed your feedback there.